### PR TITLE
Fix duplicate no_units validation error.

### PR DESCRIPTION
**Pull request by dplute**
This patch allows "no_units" to be specified for more than one measurement by giving each one a unique name in the namespace's units map.

### CODE REVIEW

**Comment by dplute** [comment]
@googlebot I signed it!

**Comment by charbull** [comment]
@dplute are we proposing to support multiple `no_units_$measurement` instead of `no_units` for any possible measurement?

**Comment by dplute** [comment]
This is meant to support the use of "no_units" for any number of measurements. (e.g. powerfactor, currentratio, voltageratio from #298)

The validator currently cannot do this because it thinks that we are trying to define a unit named "no_units" multiple times in the namespace.  This would recognize "no_units" as a special name and adjust it internally to avoid duplicate keys, which would allow "no_units" to be specified under multiple measurements without causing a duplicate definition error.

File: tools/validators/ontology_validator/yamlformat/tests/presubmit_validate_types_lib_test.py [hunk 1]
```diff
@@ -228,6 +228,28 @@ def testConfigUniverseGetUnitsForMeasurement(self):
         'discharge_fan_lost_power_alarm')
     self.assertIsNone(units)
 
+  def testConfigUniverseGetUnitsForMeasurementMultipleNoUnits(self):
+    doc = {
+      'powerfactor': [{'no_units': 'STANDARD'}],
```
Comment (charbull) [inline tools/validators/ontology_validator/yamlformat/tests/presubmit_validate_types_lib_test.py:233]: just to make this test more complete can you please use two units one standard and non-standard for powerfactor: no_units, another_one and update the test accordingly

File: tools/validators/ontology_validator/yamlformat/tests/unit_lib_test.py [hunk 2]
```diff
@@ -102,6 +102,18 @@ def testUnitFolderAddFromConfig(self):
     self.assertEqual(units['degrees_celsius'],
                      unit_lib.Unit('degrees_celsius', 'temperature', False))
 
+  def testUnitFolderAddFromConfigMultipleNoUnits(self):
+    doc = {
+      'powerfactor': [{'no_units': 'STANDARD'}],
```
Comment (charbull) [inline tools/validators/ontology_validator/yamlformat/tests/unit_lib_test.py:107]: same here test the case where a measurement has two units: no_units, another_one

**Review by charbull — CHANGES_REQUESTED** [review]
thank you, minor changes

**Comment by dplute** [comment]
@charbull I have updated the tests.

File: tools/validators/ontology_validator/yamlformat/tests/presubmit_validate_types_lib_test.py [hunk 3]
```diff
@@ -246,7 +249,7 @@ def testConfigUniverseGetUnitsForMeasurementMultipleNoUnits(self):
         unit_universe=unit_universe)
 
     units = config_universe.GetUnitsForMeasurement('powerfactor_sensor')
-    self.assertSameElements(['no_units'], units)
```
Comment (charbull) [inline tools/validators/ontology_validator/yamlformat/tests/presubmit_validate_types_lib_test.py]: can you please leave an empty line here, and remove the empty line above, 
we do follow this approach: https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/

we use blank lines to separate the parts.
//Arrange
//blank line
//Act
//blank line
//assert

**Review by charbull — APPROVED** [review]
minor tweeks just for the formatting of tests.
Thank you

File: tools/validators/ontology_validator/yamlformat/tests/presubmit_validate_types_lib_test.py [hunk 4]
```diff
@@ -228,6 +228,31 @@ def testConfigUniverseGetUnitsForMeasurement(self):
...
+
+    config_universe = presubmit_validate_types_lib.ConfigUniverse(
+        subfield_universe=None,
+        field_universe=None,
+        entity_type_universe=None,
+        state_universe=None,
+        connection_universe=None,
+        unit_universe=unit_universe)
+
+    units = config_universe.GetUnitsForMeasurement('powerfactor_sensor')
```
Comment (charbull) [inline tools/validators/ontology_validator/yamlformat/tests/presubmit_validate_types_lib_test.py:251]: line 251 to move up with the previous config_universe block
then leave empty blank line before self.assertSae...

File: tools/validators/ontology_validator/yamlformat/tests/unit_lib_test.py [hunk 5]
```diff
@@ -102,6 +102,21 @@ def testUnitFolderAddFromConfig(self):
...
+      'powerfactor': [
+        {'no_units': 'STANDARD'},
+        'another_one'
+      ],
+      'voltageratio': [{'no_units': 'STANDARD'}],
+    }
+    folder = unit_lib.UnitFolder(_GOOD_PATH)
+    folder.AddFromConfig([doc], '{0}/file.yaml'.format(_GOOD_PATH))
+    units = folder.local_namespace.units
+    self.assertEmpty(folder.GetFindings())
```
Comment (charbull) [inline tools/validators/ontology_validator/yamlformat/tests/unit_lib_test.py:116]: leave empty line before this line please

**Review by charbull — CHANGES_REQUESTED** [review]
tests formatting minor tweeks.

**Comment by dplute** [comment]
@charbull Formatting updated

**Review by charbull — APPROVED** [review]
thank you,
can you please update your branch

**Comment by dplute** [comment]
@charbull Branch updated

**Pull request merged**